stock_picking_backorder_partner_config/models/stock_picking.py

```diff
@@ -0,0 +1,78 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models
+from odoo.tools.float_utils import float_is_zero, float_compare
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _pre_action_done_hook(self):
+        if not self.env.context.get('skip_partner', False):
+            for picking in self:
+                create_partial_picking = self.get_create_partial_picking()
```

Comment:
Ouch, incorrect self reference within a for statement:
```suggestion
                create_partial_picking = picking.get_create_partial_picking()
```